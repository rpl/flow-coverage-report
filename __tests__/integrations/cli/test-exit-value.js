'use babel';

import path from 'path';

import {
  FIXTURE_PATH,
  runFlowCoverageReport
} from '../../common';

const testProjectDir = path.join(FIXTURE_PATH, 'project-low-coverage');

describe('CLI exit value', async () => {
  it('should exit with code 2 when total coverage is lower than the default threshold', async () => {
    const {exitCode, error} = await runFlowCoverageReport([
      '-i', `"src/*.js"`
    ], {cwd: testProjectDir});

    expect({exitCode, error}).toMatchSnapshot();
  });

  it('should exit with code 2 when total coverage is lower than the custom threshold', async () => {
    const {exitCode, error} = await runFlowCoverageReport([
      '-i', `"src/*.js"`,
      '--threshold', '22'
    ], {cwd: testProjectDir});

    expect({exitCode, error}).toMatchSnapshot();
  });

  it('should exit with code 0 when total coverage is higher than the custom threshold', async () => {
    const {exitCode, error} = await runFlowCoverageReport([
      '-i', `"src/*.js"`,
      '--threshold', '10'
    ], {cwd: testProjectDir});

    expect({exitCode, error}).toMatchSnapshot();
  });
});
