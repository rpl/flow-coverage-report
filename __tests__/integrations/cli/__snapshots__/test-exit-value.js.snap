// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI exit value should exit with code 0 when total coverage is higher than the custom threshold 1`] = `
Object {
  "error": null,
  "exitCode": 0,
}
`;

exports[`CLI exit value should exit with code 2 when total coverage is lower than the custom threshold 1`] = `
Object {
  "error": [Error: Command failed: flow-coverage-report.js -i "src/*.js" --threshold 22
Flow Coverage 16% is below the required threshold 22%
],
  "exitCode": 2,
}
`;

exports[`CLI exit value should exit with code 2 when total coverage is lower than the default threshold 1`] = `
Object {
  "error": [Error: Command failed: flow-coverage-report.js -i "src/*.js"
Flow Coverage 16% is below the required threshold 80%
],
  "exitCode": 2,
}
`;
